risc0/bigint2/src/ec/mod.rs

```diff
@@ -160,31 +179,71 @@ impl<const WIDTH: usize, C: Curve<WIDTH>> AffinePoint<WIDTH, C> {
         }
 
         // Return the result, based on which buffer was written to last.
-        let result_point = if result_flip { result2 } else { result1 };
-        result.buffer = result_point;
+        *result = if result_flip { result2 } else { result1 };
     }
 
+    /// Elliptic curve doubling of the affine point.
     #[stability::unstable]
     pub fn double(&self, result: &mut Self) {
         let curve = C::CURVE;
-        double_raw(self.as_u32s(), curve.as_u32s(), &mut result.buffer);
+        // If the point is zero, can short-circuit and return the identity point.
+        if self.is_zero {
+            *result = *self;
+        } else {
+            if self.buffer[1] != [0u32; WIDTH] {
```

Comment:
Is this correct? I think with double there shouldn't be a branch here, if `is_zero == false` we process with the `sys_bigint` call.

Reply:
From https://github.com/risc0/risc0/pull/2572#discussion_r1861306172

Reply:
This isn't checking for `A` is zero, this is checking for `A` is order 2, i.e. `2A = 𝒪`. This is because `-(x, y) = (x, -y)`, so we can check `A = -A` by checking `y = 0`

Reply:
See also this thread: https://github.com/risc0/risc0/pull/2572#discussion_r1861306172